example/lib/widgets/secondary_button.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+
+import '../utils/styles.dart';
+
+class SecondaryButton extends StatelessWidget {
+  const SecondaryButton({
+    Key? key,
+    required this.onPressedFunction,
+    required this.child,
+  }) : super(key: key);
+  final void Function()? onPressedFunction;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      style: ButtonStyle(
+        backgroundColor: MaterialStateProperty.resolveWith(
+          (states) {
+            if (states.contains(
+              MaterialState.disabled,
+            ))
+              return Colors.grey.shade500.withOpacity(
+                Styles.colorOpacity700,
+              );
+            return Colors.grey.shade400.withOpacity(
+              Styles.colorOpacity800,
+            );
+          },
+        ),
+        foregroundColor: MaterialStateProperty.resolveWith(
+          (states) {
+            if (states.contains(
+              MaterialState.disabled,
+            ))
+              return Colors.grey.shade700.withOpacity(
+                Styles.colorOpacity600,
+              );
+            return Colors.purple.shade700;
+          },
+        ),
+      ),
+      onPressed: onPressedFunction != null
+          ? () {
+              onPressedFunction!();
+            }
+          : null,
```

Comment:
Instead of this, use the logic that you have on location_page
hasPermission ? onPressedFunction() : (){}, try to not assign null

Reply:
Done